x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -84,7 +84,7 @@
  */
 public class LogicalPlanBuilder extends ExpressionBuilder {
 
-    private int queryDepth = 0;
+    interface PlanFactory extends Function<LogicalPlan, LogicalPlan> {}
```

Comment:
Moved this interface at the beginning since otherwise in case of merging failure, when the file gets messed up, IntelliJ gets very confused and marks the whole file full of errors. By declaring it up front, it properly picks it up and significantly reduces the errors making it easy to determine where the missing bracket is.